r/R/record-batch.R

```diff
@@ -161,7 +161,18 @@ RecordBatch$create <- function(..., schema = NULL) {
     out <- RecordBatch__from_arrays(schema, arrays)
     return(dplyr::group_by(out, !!!dplyr::groups(arrays[[1]])))
   }
-  
+
+  # If any arrays are length 1, recycle them  
+  arr_lens <- map_int(arrays, length)
```

Comment:
FWIW there is a base R function `lengths()` that does this (though I don't recall what version it was added in)

Reply:
Apparently 3.2.0.  I think in our test-r-versions job, we only got back to 3.3.0 and I can't think of anywhere else that we go back to 3.2.0 or before so will update.

Reply:
Per DESCRIPTION we require R >= 3.3 (because we depend on packages that require R >= 3.3)